### DISCUSSION TITLE
Select doesnt work right with async options

### DISCUSSION THREAD

**Discussion by s10mcow** [radix-ui/primitives#1732] [Help]
I need to have select work with a asnyc task of fetching options, but it doesnt seem to render again when the options have changed

**Comment by andy-hook** [comment]
Would you be able to provide a code sandbox example showing your usage?

**Reply by s10mcow** [reply]
https://codesandbox.io/s/ancient-fire-mh0fge?file=/src/App.js

**Reply by s10mcow** [reply]
seems i fixed my issue making a codesandbox, thanks rubber ducky :)

**Comment by s10mcow** [comment]
There is actually a different bug here, if i dont check for the value in the onValueChange callback im getting it reset with an empty string. Why is the component doing this?

![Screen Shot 2022-10-20 at 2 11 47 PM](https://user-images.githubusercontent.com/3219270/197025953-4a3139fc-3dce-4226-9484-2c43ba9a628c.png)

**Reply by s10mcow** [reply]
you can see it in the codesandbox link

https://codesandbox.io/s/ancient-fire-mh0fge?file=/src/App.js

**Reply by andy-hook** [reply]
I'm afraid i'm not following, at what point does it get reset? when the options initially resolve?
